src/main/java/duke/command/UnmarkCommand.java

```diff
@@ -0,0 +1,90 @@
+package duke.command;
+
+import java.io.IOException;
+
+import duke.exception.DukeException;
+import duke.task.TaskList;
+import duke.util.Storage;
+import duke.util.Ui;
+
+/**
+ * Encapsulates the logic for the UnmarkCommand when the user wants to unmark a certain task as
+ * not completed.
+ */
+public class UnmarkCommand extends Command {
+    private int taskNumber;
+
+    /**
+     * Constructor to create an instance of UnmarkCommand.
+     *
+     * @param taskDetails a String representation of the task descriptions
+     * @return an instance of UnmarkCommand
+     */
+    public UnmarkCommand(String taskDetails) {
+        this.taskNumber = Integer.parseInt(taskDetails);
```

Comment:
I'm not very sure if possible error that can be thrown by parseInt has been handled. For instance if user types in unmark hello when it should be unmark 1. I checked the Parser class and it does not seem to have handled this case. Perhaps can consider throwing some exception here or somewhere higher up to the chain.

Reply:
Thank you for catching something I missed out! I will be adding in a NumberFormat exception when parsing the inputs from the user to ensure that the command is correct.